### ISSUE TITLE
Cannot install: error: unknown flag `classic'

### ISSUE THREAD

**Issue by probonopd** [anbox/anbox#218]
** Please paste the result of `anbox system-info` below:**
```
Cannot install, see below
```

**Please describe your problem:**

```
me@host:~$ snap install --classic anbox-installer && anbox-installer 
error: unknown flag `classic'
```

**What were you expecting?:**

Installs

**Additional info:**

ubuntu-16.04-desktop-amd64.iso

**Comment by probonopd** [comment]
Providing an [AppImage](http://appimage.org/) would have, among others, these advantages:
- Works for most Linux distributions (including Ubuntu, Fedora, openSUSE, CentOS, elementaryOS, Linux Mint, and others)
- One app = one file = super simple for users: just download one AppImage file, [make it executable](http://discourse.appimage.org/t/how-to-make-an-appimage-executable/80), and run
- No unpacking or installation necessary
- No root needed
- No system libraries changed
- Works out of the box, no installation of runtimes needed
- Optional desktop integration with `appimaged`
- Optional binary delta updates, e.g., for continuous builds (only download the binary diff) using AppImageUpdate
- Can optionally GPG2-sign your AppImages (inside the file)
- Works on Live ISOs
- Can use the same AppImages when dual-booting multiple distributions

[Here is an overview](https://github.com/probonopd/AppImageKit/wiki/AppImages) of projects that are already distributing upstream-provided, official AppImages.

**Comment by ARCofSpada** [comment]
Did you run "sudo apt install snapd"?

**Comment by morphis** [comment]
> error: unknown flag `classic'

This simply shows you're using a pretty old snapd version. Please run

```
$ sudo apt update
$ sudo apt upgrade
```

To get your system updated to the latest available packages for your distribution.

To verify you have a recent version of snapd you can check its version with `snap --version`. This should give you `2.24` which is the latest available version released into Ubuntu 16.04.

> Providing an AppImage would have, among others, these advantages:

AppImages sadly doesn't solve our problems. We need root and we need a system service. AppImages are very well suited for standalone applications. All other advantages you listed are true for snap packages too. The only major difference is that snaps require a central management daemon on the system (snapd) which AppImages don't. One important thing which AppImages doesn't solve is the security story. I must say we don't confine the Anbox snap yet but that will happen in the future to isolate it from the rest of the system quite a bit more.

**Comment by meki383** [comment]
notice that you are writing the wrong command
(sudo snap install -classic anbox-installer && anbox-installer )
but it is so necessary
sudo snap install --classic anbox-installer && anbox-installer
